phoenix-core/src/test/java/org/apache/phoenix/cache/ServerMetadataCacheImplTest.java

```diff
@@ -121,7 +127,7 @@ public void testCacheForBaseTable() throws Exception {
             createTable(conn, tableNameStr, NEVER);
             pTable = PhoenixRuntime.getTableNoCache(conn,
                     tableNameStr);// --> First call to CQSI#getTable
-            ServerMetadataCache cache = ServerMetadataCache.getInstance(config);
+            ServerMetadataCacheTestImpl cache = ServerMetadataCacheTestImpl.getInstance(config, serverName);
```

Comment:
Instead of directly calling ServerMetadataCacheTestImpl, we should call it via findCoprocessor method something like this:
```
private ServerMetadataCache getCache() {
  String phoenixRegionServerEndpoint = conf.getValue(REGIONSERVER_COPROCESSOR_CONF_KEY);
  assertNotNull(phoenixRegionServerEndpoint);
  PhoenixRegionServerEndpoint coproc = regionServer.getRegionServerCoprocessorHost()               
  .findCoprocessor(phoenixRegionServerEndpoint.class); 
  assertNotNull(coproc);
  ServerMetadataCache cache = coproc.getServerMetadataCache();
  assertNotNull(cache);
  return cache;
}
```